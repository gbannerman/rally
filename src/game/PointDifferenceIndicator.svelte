<script lang="ts">
  export let pointDifference: number[];
</script>

<div class="PointDifferenceIndicator">
  {#if pointDifference[0] > 0}
    <span class="player1" aria-label="Player 1 Point Difference"
      >+{pointDifference[0]}</span
    >
  {/if}
  {#if pointDifference[1] > 0}
    <span class="player2" aria-label="Player 2 Point Difference"
      >+{pointDifference[1]}</span
    >
  {/if}
</div>

<style>
  .PointDifferenceIndicator {
    position: absolute;
    bottom: calc(35%);
    left: calc(50% - 50px);
    width: 100px;
    display: grid;
  }

  span {
    font-size: x-large;
  }

  .player1 {
    color: #5b0b05;
    justify-self: left;
  }

  .player2 {
    color: #112f70;
    justify-self: right;
  }
</style>
