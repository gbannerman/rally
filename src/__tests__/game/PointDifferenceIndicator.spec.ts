import { render, fireEvent } from "@testing-library/svelte";

import PointDifferenceIndicator from "../../game/PointDifferenceIndicator.svelte";

describe("PointDifferenceIndicator component", () => {
  it("renders correctly", () => {
    const { container } = render(PointDifferenceIndicator, {
      pointDifference: [7, -7],
    });

    expect(container).toMatchSnapshot();
  });

  it("does not display point difference when even score", async () => {
    const { queryByLabelText } = render(PointDifferenceIndicator, {
      pointDifference: [0, 0],
    });

    const player1PointDifference = queryByLabelText(
      "Player 1 Point Difference"
    );
    const player2PointDifference = queryByLabelText(
      "Player 2 Point Difference"
    );

    expect(player1PointDifference).not.toBeInTheDocument;
    expect(player2PointDifference).not.toBeInTheDocument;
  });

  it("shows player 1 score when leading", async () => {
    const { getByLabelText, queryByLabelText } = render(
      PointDifferenceIndicator,
      {
        pointDifference: [2, -2],
      }
    );

    const player1PointDifference = getByLabelText("Player 1 Point Difference");
    const player2PointDifference = queryByLabelText(
      "Player 2 Point Difference"
    );

    expect(player1PointDifference).toHaveTextContent("+2");
    expect(player2PointDifference).not.toBeInTheDocument;
  });

  it("shows player 2 score when leading", async () => {
    const { getByLabelText, queryByLabelText } = render(
      PointDifferenceIndicator,
      {
        pointDifference: [-12, 12],
      }
    );

    const player1PointDifference = queryByLabelText(
      "Player 1 Point Difference"
    );
    const player2PointDifference = getByLabelText("Player 2 Point Difference");

    expect(player1PointDifference).not.toBeInTheDocument;
    expect(player2PointDifference).toHaveTextContent("+12");
  });
});
