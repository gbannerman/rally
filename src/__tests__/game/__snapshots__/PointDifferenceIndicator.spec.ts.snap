// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointDifferenceIndicator component renders correctly 1`] = `
<body>
  <div>
    <div
      class="PointDifferenceIndicator svelte-16qa96h"
    >
      <span
        aria-label="Player 1 Point Difference"
        class="player1 svelte-16qa96h"
      >
        +
        7
      </span>
       
    </div>
  </div>
</body>
`;
